Antidote/AbstractCallViewController.h

```diff
@@ -18,21 +16,20 @@
 + (instancetype)new NS_UNAVAILABLE;
 
 /**
- * Create an instance of the CallViewController.
- * @param chat Appropriate chat for the call.
- * @param manager The call maanger for the application.
+ * Create an instance of the AbstractCallViewController.
+ * @param nickname The nickname associated with the caller.
  */
-- (instancetype)initWithCall:(OCTCall *)call submanagerCalls:(OCTSubmanagerCalls *)manager;
+- (instancetype)initWithCallerNickname:(NSString *)nickname;
 
 /**
- * The call associated with this view controller.
+ * Abstract delegate
  */
-@property (strong, nonatomic, readonly) OCTCall *call;
+@property (weak, nonatomic) id delegate;
```

Comment:
Do we need this delegate without protocol? I think you can remove it and leave only concrete delegates of controllers.